extensions/rocksdb-repos/RocksDatabase.cpp

```diff
@@ -0,0 +1,139 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "RocksDatabase.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace internal {
+
+OpenRocksDB::OpenRocksDB(gsl::not_null<RocksDatabase*> db, gsl::not_null<std::shared_ptr<rocksdb::DB>> impl) : db_(std::move(db)), impl_(std::move(impl)) {}
+
+rocksdb::Status OpenRocksDB::Put(const rocksdb::WriteOptions& options, const rocksdb::Slice& key, const rocksdb::Slice& value) {
+  rocksdb::Status result = impl_->Put(options, key, value);
+  if (result == rocksdb::Status::NoSpace()) {
+    db_->invalidate();
+  }
+  return result;
+}
+
+rocksdb::Status OpenRocksDB::Get(const rocksdb::ReadOptions& options, const rocksdb::Slice& key, std::string* value) {
+  rocksdb::Status result = impl_->Get(options, key, value);
+  if (result == rocksdb::Status::NoSpace()) {
+    db_->invalidate();
+  }
+  return result;
+}
+
+std::vector<rocksdb::Status> OpenRocksDB::MultiGet(const rocksdb::ReadOptions& options, const std::vector<rocksdb::Slice>& keys, std::vector<std::string>* values) {
+  std::vector<rocksdb::Status> results = impl_->MultiGet(options, keys, values);
+  for (const auto& result : results) {
+    if (result == rocksdb::Status::NoSpace()) {
+      db_->invalidate();
+    }
+  }
```

Comment:
Could the db be invalidated on the first bad result only?

```suggestion
    if (std::any_of(results.begin(), results.end(), [](const rocksdb::Status& status){ return status == rocksdb::Status::NoSpace();})
      db_->invalidate();
```

Reply:
nice catch (although `invalidate` is idempotent), I just added a `break`, because the proposed solution is too verbose for my liking, I dream of a day when we can do something like `results.some(_ == rocksdb::Status::NoSpace())` but that will probably come in `C++never`

Reply:
I think 
```c++
auto is_nospace = [](auto status){ return status == rocksdb::Status::NoSpace(); };
if (std::ranges::any_of(results, is_nospace) { ... }
```
is OK, and it is reality in C++20, so it will be available in minificpp in about ten years. :)